sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -480,6 +486,76 @@ def test_with_input_types_decorator_violation(self):
         pc = p | Create(l_3_tuple)
         _ = pc | beam.CombineGlobally(self.fn)
 
+#
+# Test cases for streaming.
+#
+@attr('ValidatesRunner')
+class TimestampCombinerTest(unittest.TestCase):
+
+  class RecordFn(beam.DoFn):
+    def process(self, elm=beam.DoFn.ElementParam, ts=beam.DoFn.TimestampParam):
+      yield (elm, ts)
+
+  @unittest.skip('BEAM-8657')
+  def test_combiner_earliest(self):
+    """Test TimestampCombiner with EARLIEST."""
+    options = PipelineOptions(streaming=True)
+    with TestPipeline(options=options) as p:
+      result = (p
+                | 'main TestStream' >> TestStream()
+                .add_elements([window.TimestampedValue(('k', 100), 2)])
+                .add_elements([window.TimestampedValue(('k', 400), 7)])
+                .advance_watermark_to_infinity()
+                | 'main windowInto' >> beam.WindowInto(
+                    window.FixedWindows(10),
+                    timestamp_combiner=TimestampCombiner.OUTPUT_AT_EARLIEST)
+                | 'Combine' >> beam.CombinePerKey(sum))
+
+      records = (result | beam.ParDo(self.RecordFn()))
```

Comment:
As before, a full DoFn is not needed here, just write `beam.Map(lambda e, ts=beam.DoFn.TimestampParam: (e, ts))`

Reply:
Also, no need for the surrounding ()'s.

Reply:
Ah. Thanks for the tips. :-D